src/main/java/net/starlark/java/eval/Starlark.java

```diff
@@ -148,7 +148,11 @@ public static boolean isImmutable(Object x) {
    * @throws EvalException otherwise.
    */
   public static void checkHashable(Object x) throws EvalException {
-    if (x instanceof StarlarkValue) {
+    if (x instanceof String) {
+      // Strings are the most common dict keys.
+      // Check them first for that reason, but also because
+      // instanceof interface if very slow.
+    } else if (x instanceof StarlarkValue) {
       ((StarlarkValue) x).checkHashable();
     } else {
       Starlark.checkValid(x);
```

Comment:
Nit: since we're handling the string case above, remove "String and" from the comment here.

Reply:
Done.